### ISSUE TITLE
Integer division by zero

### ISSUE THREAD

**Issue by pauldreik** [stephenberry/glaze#367]
The following code (to be inserted into [json_test.cpp](https://github.com/stephenberry/glaze/blob/12e75bd44f55256ca9471040e1064b610213a400/tests/json_test/json_test.cpp#L238)) causes an integer division by zero which is UB:
```cpp
   "demonstrate division by zero"_test = [] {
      std::string_view buffer{R"({"i":0.00666666666666666600})"};
      my_struct value{};
      glz::context ctx{};
      [[maybe_unused]] const auto ec = glz::read<glz::opts{.force_conformance = false}>(value, buffer, ctx);
   };
```
The division happens here:
https://github.com/stephenberry/glaze/blob/12e75bd44f55256ca9471040e1064b610213a400/include/glaze/util/strod.hpp#L664
where exp is -20. That element of powers_of_ten_int is however zero, because there is one element too little here:
https://github.com/stephenberry/glaze/blob/12e75bd44f55256ca9471040e1064b610213a400/include/glaze/util/strod.hpp#L50

Adding the missing element does not work, because it exceeds what fits in a 64 bit int.

**Comment by stephenberry** [comment]
@pauldreik Thanks so much for this.

**Comment by stephenberry** [comment]
Test has been added, and a fix was made in #371
